src/eval.c

```diff
@@ -264,15 +264,27 @@ void scriptingInit(int setup) {
     lctx.lua = lua;
 }
 
+/* Free lua_scripts dict and close lua interpreter. */
+void freeLuaScriptsSync(dict *lua_scripts, lua_State *lua) {
+    dictRelease(lua_scripts);
+    lua_close(lua);
+
+#if defined(__linux__) && !defined(__APPLE__) && !defined(USE_LIBC)
+    /* The lua interpreter may hold a lot of memory internally, and lua is
+     * using libc. libc may take a bit longer to return the memory to the OS,
+     * so after lua_close, we call malloc_trim try to purge it earlier.
+     * We only do this if redis and lua use different allocators. */
+    malloc_trim(0);
```

Comment:
is it a good idea to implement `zmalloc_purge()`?
use different purge methods for different allocater.

Reply:
i don't think so.
in this case we're not purging Redis's allocator. we're purging the "other" one.
we can wrap it like we do for `zlibc_free`